lib/PermissionManager.php

```diff
@@ -47,16 +53,24 @@ private function splitGroups($groupString) {
 		return explode('|', $groupString);
 	}
 
-	public function isEnabledForUser(IUser $user) {
+	public function isEnabledForUser(string $userId = null) {
+		if ($userId === null) {
+			$user = $this->userSession->getUser();
+			$userId = $user ? $user->getUID() : null;
+		}
+
+		if ($userId === null) {
+			return true;
+		}
+
 		$enabledForGroups = $this->config->getAppValue(Application::APPNAME, 'use_groups', '');
 		if ($enabledForGroups === '') {
 			return true;
 		}
 
 		$groups = $this->splitGroups($enabledForGroups);
-		$uid = $user->getUID();
 		foreach ($groups as $group) {
-			if ($this->groupManager->isInGroup($uid, $group)) {
+			if ($this->groupManager->isInGroup($userId, $group)) {
```

Comment:
Wondering if we could use `IAppManager::isEnabledForUser` instead

Reply:
Richdocuments has its own setting as we need to load the app in early stages to make sure that preview generation can register and act on file system operations. That will disable the server built-in app limitation based on groups. Also this would block the app bootstrap for guests.

https://github.com/nextcloud/server/blob/66ffedcb1662088007711d132df7a630a313c572/lib/private/App/AppManager.php#L232-L233
https://github.com/nextcloud/server/blob/66ffedcb1662088007711d132df7a630a313c572/lib/private/App/AppManager.php#L56-L65

Reply:
That makes sense, thanks for explaining :)

PR looks good then!